hotrod/services/frontend/Sources/API/HTTPServerHandler.swift

```diff
@@ -0,0 +1,149 @@
+        let span = InstrumentationSystem.tracer.startSpan(
+            named: operationName,
+            baggage: context.baggage,
+            ofKind: .server
+        )
+        span.attributes.http.method = head.method.rawValue
+        span.attributes.http.flavor = "\(head.version.major).\(head.version.minor)"
+        span.attributes.http.target = head.uri
+        span.attributes.http.serverRoute = head.uri
+        span.attributes.http.serverClientIP = context.remoteAddress?.ipAddress
+        span.attributes.net.peerIP = context.remoteAddress?.ipAddress
+        span.attributes.http.userAgent = head.headers.first(name: "user-agent")
+        if let localAddress = context.localAddress, let port = localAddress.port {
+            span.attributes.net.hostPort = port
+            if let host = localAddress.ipAddress {
+                span.attributes.http.host = "\(host):\(port)"
+                span.attributes.http.serverName = host
+            }
+        }
+        let request = Request(head: head, span: span)
+        self.logger.info("\(operationName)")
+        self.state.requestReceived(request)
+    }
+
+    private func completeResponse(context: ChannelHandlerContext) {
+        context.fireChannelReadComplete()
+
+        let request = self.state.requestComplete()
+        let responseStatus = HTTPResponseStatus.ok
+
+        let messageBuffer = context.channel.allocator.buffer(staticString: "Hello\n")
+        let responseHead = HTTPResponseHead(version: request.head.version, status: responseStatus, headers: [
+            "Content-Type": "text/plain",
+            "Content-Length": "\(messageBuffer.readableBytes)",
+        ])
+
+        request.span.attributes.http.statusCode = Int(responseStatus.code)
+        request.span.attributes.http.statusText = responseStatus.reasonPhrase
+        request.span.attributes.http.responseContentLength = messageBuffer.readableBytes
+
+        self.recordMetrics(forRequest: request, respondedWith: responseStatus)
+
+        context.eventLoop.scheduleTask(in: .seconds(2)) {
+            context.write(self.wrapOutboundOut(.head(responseHead)), promise: nil)
+            context.write(self.wrapOutboundOut(.body(.byteBuffer(messageBuffer))), promise: nil)
+            self.state.responseComplete(context: context, endData: self.wrapOutboundOut(.end(nil)))
+        }
+    }
+
+    private func recordMetrics(forRequest request: Request, respondedWith status: HTTPResponseStatus) {
+        let dimensions = [
+            ("method", request.head.method.rawValue),
+            ("path", request.head.uri),
+            ("status", "\(status.code)"),
+        ]
+
+        Counter(label: "http_requests_total", dimensions: dimensions).increment()
+
+        Timer(
+            label: "http_request_duration_seconds",
+            dimensions: dimensions,
+            preferredDisplayUnit: .seconds
+        ).recordNanoseconds(DispatchTime.now().uptimeNanoseconds - request.startTime.uptimeNanoseconds)
```

Comment:
added to the ticket https://github.com/apple/swift-metrics/issues/79 that sugar would be nice